huaweicloud/services/cae/data_source_huaweicloud_cae_environments.go

```diff
@@ -0,0 +1,196 @@
+package cae
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API CAE GET /v1/{project_id}/cae/environments
+func DataSourceCaeEnvironments() *schema.Resource {
```

Comment:
```suggestion
func DataSourceEnvironments() *schema.Resource {
```
There is unnecessary to contain the package name in your function names.

Reply:
done